### PR TITLE
New Portal Landing Page

### PR DESCRIPTION
**Pull request by kmpaul**
As promised, here is the new landing page for the Portal.  I've moved things around to accommodate the new page structure, and that means I had to develop some new text...and I didn't change some old text.  It might be that the old text needs to be updated, too.

So, **please look closely** at the site and all of the contents.  Read the text and determine if it makes sense and works with the new layout.  Click the links and make sure they work and take you to the right place.  Please look at the formatting and decide if it looks clean, the design is good, etc.

### CODE REVIEW

**Comment by kmpaul** [comment]
Ok, @ProjectPythia/infrastructure folks!  This is the new theme.  Please take a close look at it and let me know what you think.

**Comment by kmpaul** [comment]
CC @ProjectPythia/core @ProjectPythia/education

**Comment by brian-rose** [comment]
@kmpaul congratulations! I think this looks fantastic. I clicked around to check most of the links and inspect the look and feel. I didn't see any errors.

I have a  few comments about content:
- I think the paragraph under the "Project Pythia" heading is too verbose, seeing as it's the first thing people see when they hit the page. I think we can come up something snappier that describes the project in a few short catchphrases, and leave the details to the "About" page
- The "About" page needs updating to better reflect where we are and where we are aiming with our content.
- We recently pushed out a much-revised [Contributors Guide](https://projectpythia.org/pages/contributing.html) on the portal. Can we wrap this in to the new portal?
- I suggest a more prominent link to the Code of Conduct document, it's currently buried pretty deep. Perhaps a third button in the "Contributing" subheading under "Join Us!", along with a mention in that bit of text?

**Review by andersy005 — COMMENTED** [review]
This looks fantastic, @kmpaul! I have one comment. On smaller screen sizes, the order of sentences in the `copyright` section is broken:


Screen Shot 2021-06-18 at 7 09 10 AM

Have you seen this?

**Comment by jukent** [comment]
Some notes on the gallery:
- I would like the gallery card tags to be clickable without opening the modal first. 
- I am curious to see the version where the cards are the same size. I know you said it looked bad but I can't picture it looking that bad (is this because there is no "open details" button")
- Is the hope that a horizontal card stacking will remove the awkwardness of the cards
- I can address all of this in a separate PR

**Comment by dcamron** [comment]
Would we look to pull the local theme back out to its repo after some time?

**Comment by kmpaul** [comment]
@jukent Im confused about your comment that the cards are not the same size. The demo has the cards all the same size. Can you not see it?

**Comment by kmpaul** [comment]
@dcamron Yes. I hope to pull the theme out into its own repo. In another PR.

**Comment by kmpaul** [comment]
@jukent As for making the cards clickable without opening the modal, you would need to put something like the "See Details" button back. But I found that cluttered the cards, so maybe there is a better way of doing it.

**Comment by kmpaul** [comment]
@jukent Also, yes, I thought the horizontal cards (`w-100`) would make this easier to do.

**Comment by kmpaul** [comment]
@brian-rose:

> I have a few comments about content:
> 
> * I think the paragraph under the "Project Pythia" heading is too verbose, seeing as it's the first thing people see when they hit the page. I think we can come up something snappier that describes the project in a few short catchphrases, and leave the details to the "About" page

Yes.  I thought the same, but I didn't want to change content with this PR.  I really just wanted to move content, if necessary...and minimally add content where necessary.

> * The "About" page needs updating to better reflect where we are and where we are aiming with our content.

Agreed.  Perhaps that could be done in a new PR, too.

> * We recently pushed out a much-revised [Contributors Guide](https://projectpythia.org/pages/contributing.html) on the portal. Can we wrap this in to the new portal?

Yes.  I will copy that over to the new portal and rebuild for people to see.

> * I suggest a more prominent link to the Code of Conduct document, it's currently buried pretty deep. Perhaps a third button in the "Contributing" subheading under "Join Us!", along with a mention in that bit of text?

That's a good idea.  I can add that, too.

**Comment by kmpaul** [comment]
@brian-rose:  So, the way the old portal was built, it copied the `/CONTRIBUTING.md` file into the `/content/pages/` directory.  Are we no longer going to have a `/CONTRIBUTING.md` file?  If was _are_ going to keep one, then we need to make them consistent.  ...So, are we going to copy `/CONTRIBUTING.md` into `/content/pages/`?  Or are we going to copy the one in `/content/pages/` into `/`?  (Normally, there is a `CONTRIBUTING.md` file.)

**Comment by kmpaul** [comment]
Ok, summarizing the comments so far:

- [ ] @brian-rose: I think the paragraph under the "Project Pythia" heading is too verbose, seeing as it's the first thing people see when they hit the page. I think we can come up something snappier that describes the project in a few short catchphrases, and leave the details to the "About" page

  - @kmpaul: I think we should handle this in a future PR, once the technical aspects of the new theme are in place with this PR.

- [ ] @brian-rose: The "About" page needs updating to better reflect where we are and where we are aiming with our content.

  - @kmpaul: I agree, and I think that should be in another PR.

- [ ] @brian-rose: We recently pushed out a much-revised [Contributors Guide](https://projectpythia.org/pages/contributing.html) on the portal. Can we wrap this in to the new portal?

  - @kmpaul:  Yes.  I can do that, but I want clarification on whether the `/CONTRIBUTING.md` file should be updated or removed and which file (`/CONTRIBUTING.md` vs `/content/pages/contributing.md`) should be kept (and which one is a copy created at construction time).

- [ ] @brian-rose: I suggest a more prominent link to the Code of Conduct document, it's currently buried pretty deep. Perhaps a third button in the "Contributing" subheading under "Join Us!", along with a mention in that bit of text?

  - @kmpaul: I think I have added this feature with a recent commit.  Please check it out in the RTD demo.

- [ ] @andersy005: On smaller screen sizes, the order of sentences in the copyright section is broken

  - @kmpaul:  I believe this is fixed, now.

- [ ] @jukent: I would like the gallery card tags to be clickable without opening the modal first.

  - @kmpaul:  That's fair.  Perhaps we can figure out a way of adding that feature in a future PR (as you suggested).

- [ ] @jukent: I am curious to see the version where the cards are the same size. I know you said it looked bad but I can't picture it looking that bad (is this because there is no "open details" button")

  - @kmpaul:  Please confirm that the RTD demo ([here]()) shows the cards all the same size.

- [ ] @jukent: Is the hope that a horizontal card stacking will remove the awkwardness of the cards

  - @kmpaul: I was thinking that.  Perhaps we can try that in a future PR that adds the Bootstrap `w-100` class to the cards and/or the `col-12` class to the column `div`s of the gallery.  Then we can add more content and lay it out differently.

- [ ] @dcamron:  Would we look to pull the local theme back out to its repo after some time?

  - @kmpaul:  Yes, in a future PR.  I figure once the theme has settled down with the existing portal content, then we can pull it out and make it a separate package (in the `ProjectPythia/sphinx-pythia-theme` repository).

**Please check off the items that you feel are satisfactorily answered, and add more comments to address the ones that are not.**

**Comment by brian-rose** [comment]
> @brian-rose: So, the way the old portal was built, it copied the `/CONTRIBUTING.md` file into the `/content/pages/` directory. Are we no longer going to have a `/CONTRIBUTING.md` file? If we _are_ going to keep one, then we need to make them consistent. ...So, are we going to copy `/CONTRIBUTING.md` into `/content/pages/`? Or are we going to copy the one in `/content/pages/` into `/`? (Normally, there is a `/CONTRIBUTING.md` file.)

Let me check up on this. @clyne led the recent effort to consolidate the contributors guides across our different repos. As he is on PTO now, I'll see if I can figure out the organizational structure.

**Comment by clyne** [comment]
> > @brian-rose: So, the way the old portal was built, it copied the `/CONTRIBUTING.md` file into the `/content/pages/` directory. Are we no longer going to have a `/CONTRIBUTING.md` file? If we _are_ going to keep one, then we need to make them consistent. ...So, are we going to copy `/CONTRIBUTING.md` into `/content/pages/`? Or are we going to copy the one in `/content/pages/` into `/`? (Normally, there is a `/CONTRIBUTING.md` file.)
> 
> Let me check up on this. @clyne led the recent effort to consolidate the contributors guides across our different repos. As he is on PTO now, I'll see if I can figure out the organizational structure.

There is a CONTRIBUTING.md file in the foundations and landing page repo (and probably should be one in the data repo as well). This document simply provides a link back to the contributing.html document for each site. At lease it did. Things seem to have changed since I overhauled the contributor's information a week ago... Not sure what happened, and I'm headed out the door for the week.

**Comment by brian-rose** [comment]
Following up on @clyne's comments:
- The main portal houses a beginner-friendly guide to getting involved that serves all of Project Pythia. 
    - Source: https://github.com/ProjectPythia/projectpythia.github.io/blob/main/content/pages/contributing.md
    - Published here: https://projectpythia.org/pages/contributing.html
- **I think this is the file we want to link to from the "Contributor's Guide" button on the new portal**
- We *also* have repo-specific `/CONTRIBUTING.md` in two places
    - Portal repo: https://github.com/ProjectPythia/projectpythia.github.io/blob/main/CONTRIBUTING.md
    - Foundations repo: https://github.com/ProjectPythia/pythia-foundations/blob/main/CONTRIBUTING.md
- These files are *not* meant to be rendered into either site.
- The Foundations site has its own repo-specific detailed guide. The Foundations `/CONTRIBUTING.md` just contains a link to that page.

I think our current disconnect is the *for the portal site only*, we currently have the detailed technical instructions for building and rendering the site ONLY in the `/CONTRIBUTING.md` file, with the intention that this is viewed on GitHub rather than rendered into our published site. The reason for this is that the published Contributor's Guide is meant to be a high-level guide applicable across all our repos. However this currently leaves us with a (possibly confusing) asymmetry between the Portal and the Foundations repos. 

I think we could have a longer discussion about the organizational strategy. But for the sake of moving this new portal forward, this is my suggestion:
- Keep the [latest existing CONTRIBUTING.md](https://github.com/ProjectPythia/projectpythia.github.io/blob/main/CONTRIBUTING.md) in place at the root of the repo. This contains a link to the Guide published on the Portal
- Publish the [new Contributor's Guide](https://github.com/ProjectPythia/projectpythia.github.io/blob/main/content/pages/contributing.md) on the Portal

tl;dr there should be `/CONTRIBUTING.md` with technical details, but it is NOT copied into the rendered site because we have different content specifically designed for that.

**Comment by kmpaul** [comment]
@brian-rose @clyne:  Sounds good.  I'll get the latest Contributor's Guide into the new portal.

**Comment by brian-rose** [comment]
> @brian-rose @clyne: Sounds good. I'll get the latest Contributor's Guide into the new portal.

Ok, I think this is resolved for now.

**Review by andersy005 — DISMISSED** [review]

**Comment by kmpaul** [comment]
@jukent:  I've tried to address your concerns in the latest commits.  Let me know what you think.  I think it's a decent first pass.

**Comment by jukent** [comment]
> @jukent Im confused about your comment that the cards are not the same size. The demo has the cards all the same size. Can you not see it?

Oh I was referring to the demo you gave during the meeting last Thursday. In the build now it looks like the cards are all the same width, horizontally stacked, without modals, and the tag button works. I think it looks great!

**Comment by kmpaul** [comment]
@jukent Actually, some of the modals are there.  😄  Look for some of the cards that have a "...more" at the end of their description.  These descriptions were cut off because of length and I added the "...more" link to open the modal.  But these are the only modals that exist, now.

**Comment by jukent** [comment]
> @jukent Actually, some of the modals are there. 😄 Look for some of the cards that have a "...more" at the end of their description. These descriptions were cut off because of length and I added the "...more" link to open the modal. But these are the only modals that exist, now.

I see that now! Looks great and makes sense. It's also nice knowing that the modal functionality is still there since we worked so hard on it haha (and if we decide to add more content later we still have the ability to put it in a modal easily).

**Review by jukent — APPROVED** [review]

**Comment by brian-rose** [comment]
@kmpaul I'm giving a talk on Pythia tomorrow for the [Pangeo Showcase](https://pangeo.io/pangeo-showcase.html#pangeo-showcase) series. Another opportunity to get a bunch of eyes on our sites! 

As I'm planning my talk, just wondering: do you anticipate being able to merge the new portal in the next 24 hours, or still working through details?

**Review by andersy005 — APPROVED** [review]

**Comment by kmpaul** [comment]
@brian-rose:  Now that I have 2 approvals, I'll merge it now.
